spec/features/adoption_application/show_spec.rb

```diff
@@ -0,0 +1,260 @@
+
+      expect(page).to have_content("Add Pets to this Application")
+      expect(page).to have_content("Search Pet by Name")
+
+      fill_in "search", with: "sc"
+      click_on "Search"
+
+      within "#pet-#{pet_1.id}" do
+         expect(page).to have_content(pet_1.name)
+         expect(page).to have_content(pet_1.breed)
+         expect(page).to have_content(pet_1.age)
+         expect(page).to have_content(pet_1.adoptable)
+         expect(page).to have_button("#{pet_1.name}")
+      end
+      
+      within "#pet-#{pet_2.id}" do
+         expect(page).to have_content(pet_2.name)
+         expect(page).to have_content(pet_2.breed)
+         expect(page).to have_content(pet_2.age)
+         expect(page).to have_content(pet_2.adoptable)
+         expect(page).to have_button("#{pet_2.name}")
+      end
+   end
+
+   it 'display a button to adopt the pet under each pet' do
+      shelter = Shelter.create(name: "Mystery Building", city: "Irvine CA", foster_program: false, rank: 9)
+      pet_1 = Pet.create(name: "Scooby", age: 2, breed: "Great Dane", adoptable: true, shelter_id: shelter.id)
+      pet_2 = Pet.create(name: "Scrappy", age: 1, breed: "Pit Mix", adoptable: true, shelter_id: shelter.id)
+      application = AdoptionApplication.create!(name: "Mel", street_address: "23 Main St", city: "Denver", state: "CO", zip_code: 80303, description: "I have a fenced backyard and love dogs")
+
+      visit "/applications/#{application.id}"
+
+      fill_in "search", with: "scooby"
+      click_button "Search"
+
+      within "#pet-#{pet_1.id}" do
+         expect(page).to have_content(pet_1.name)
+         expect(page).to have_content(pet_1.breed)
+         expect(page).to have_content(pet_1.age)
+         expect(page).to have_content(pet_1.adoptable)
+         expect(page).to have_button("#{pet_1.name}")
+         expect(page).to have_button("Adopt Scooby")
+      end
+      
+   end
+
+   it 'shows added pets to adoption application' do
+      shelter = Shelter.create(name: "Mystery Building", city: "Irvine CA", foster_program: false, rank: 9)
+      pet_1 = Pet.create(name: "Scooby", age: 2, breed: "Great Dane", adoptable: true, shelter_id: shelter.id)
+      pet_2 = Pet.create(name: "Scrappy", age: 1, breed: "Pit Mix", adoptable: true, shelter_id: shelter.id)
+      application = AdoptionApplication.create!(name: "Mel", street_address: "23 Main St", city: "Denver", state: "CO", zip_code: 80303, description: "I have a fenced backyard and love dogs")
+
+      visit "/applications/#{application.id}"
+
+      fill_in "search", with: "scooby"
+      click_button "Search"
+
+      click_button "Adopt Scooby"
+      
+      fill_in "search", with: "a"
+
+      expect(page).to have_content(pet_1.name)
+   end
+
+   it "shows a session to why the user would be a good owner after they add the pet to the application" do 
+      
+      shelter = Shelter.create(name: "Mystery Building", city: "Irvine CA", foster_program: false, rank: 9)
```

Comment:
Great feature testing! For these next several tests that have some common test data, consider putting it in a before :each block, even if it's just for the tests in a describe block rather than for all tests in this file